api/app.go

```diff
@@ -306,17 +313,53 @@ func (app *App) AddError() {
 	app.Errors.Update(1)
 }
 
+func buildAddress(port int) string {
+	if port == 0 {
+		return "localhost:"
```

Comment:
I think `localhost:0` does what you want it to, no need to special case it.

In any case, I'd suggest dropping this function and doing `fmt.Sprintf("%v:%v", host, port)` where you  need  it - buildAddress doesn't pull its own weight imo.

Reply:
buildAddress removed.